### PR TITLE
fix: fix undefined error logs

### LINKED ISSUE #5135: [Task]: Fix `undefined` errors

### PR DESCRIPTION
**Pull request by MarkNerdi996**
## Summary

This PR fixes the issue that a lot of errors get logged as `undefined`. Further we now use the `handleError` on all relevant places.

## Changelog

```
- replace `new BaseError()` with `handleError`
- fix `undefined` error by passing in the correct attributes to `handleError`
- cleanup 
```

## Relevant Issues

Closes #5135

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by Tuditi — COMMENTED** [review]
Left two comments :relaxed:

File: packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte [hunk 1]
```diff
@@ -103,8 +103,8 @@
     async function handleMint(): Promise<void> {
         try {
             await checkActiveProfileAuth(mintAction, { stronghold: true, ledger: false })
-        } catch (reason) {
-            handleError(reason.error)
+        } catch (error) {
```
[the remark this answers is not in the document]
Comment (MarkNerdi) [reply packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte:106]: yes, unifying it to  `err` would be fine with me. `reason` was just to misleading for me

File: packages/shared/components/popups/send/SendConfirmationPopup.svelte [hunk 2]
```diff
@@ -153,7 +153,7 @@
             }
             await checkActiveProfileAuth(sendOutputAndClosePopup, { stronghold: true, ledger: false })
         } catch (err) {
-            handleError(err?.error)
+            handleError(err)
```
[the remark this answers is not in the document]
Comment (MarkNerdi) [reply packages/shared/components/popups/send/SendConfirmationPopup.svelte:156]: nice catch

File: packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicole-obrien) [reply packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte]: But reason is the name of the parameter passed to catch, and err is not following conventions, and we cant have error as we have sub properties and cannot do error.error.

[the remark this answers is not in the document]
Comment (Tuditi) [reply packages/shared/components/popups/MintNativeTokenConfirmationPopup.svelte]: We use `catch (err)` in 87 places vs `catch (error)` in 23 places vs `catch (reason)` in 5 places. I think it's okay to make our own convention of having `catch (err)` at this point.

**Pull request merged**